src/decimal.rs

```diff
@@ -771,6 +775,12 @@ impl Decimal {
                     _ => {}
                 }
             }
+            RoundingStrategy::RoundUp => {
+                if decimal_portion != [0, 0, 0] {
```

Comment:
For consistency, we should probably use `is_all_zero`
```suggestion
                if is_all_zero(&decimal_portion) {
```

Reply:
Ah sorry - I got the logic for this wrong. It should be `!is_all_zero`!